src/wallet/wallet.cpp

```diff
@@ -2325,7 +2341,11 @@ void CWallet::DecrementNoteWitnesses(const CBlockIndex* pindex)
     assert(nWitnessCacheSize > 0);
 
     // ORCHARD: rewind to the last checkpoint.
-    orchardWallet.RewindToLastCheckpoint();
+    if (consensus.NetworkUpgradeActive(pindex->nHeight, Consensus::UPGRADE_NU5)) {
```

Comment:
We need a test to ensure that in a reorg-across-activation-height scenario, we don't end up in an inconsistent state for anyone who happens to receive Orchard notes in the first few blocks after NU5 activation.